src/world/action/sneaky_parasol.c

```diff
@@ -1,52 +1,368 @@
 #include "common.h"
 #include "npc.h"
+#include "effects.h"
 
 typedef struct SneakyParasolUnk {
-    /* 0x00 */ Vec3f unk_00;
+    /* 0x00 */ Npc* npc;
+    /* 0x04 */ s32 unk_04;
+    /* 0x08 */ s32 unk_08;
     /* 0x0C */ s32 unk_0C;
-    /* 0x10 */ s32 unk_10;
-    /* 0x14 */ s32 unk_14;
-    /* 0x18 */ s32 unk_18;
+    /* 0x10 */ Vec3f position;
     /* 0x1C */ f32 unk_1C;
     /* 0x20 */ f32 unk_20;
 } SneakyParasolUnk;
 
 extern SneakyParasolUnk D_802B6E80;
 
-f64 fabs(f64 f);
+Npc *func_802B6000_E2A6B0(void)
+{
+  PlayerStatus *playerStatus = &gPlayerStatus;
```

Comment:
The pointer should be next to PlayerStatus. If you're using mips-to-c, you can do `--pointer-style left`

Reply:
(applies below too)